libraries/AP_HAL_Linux/Util.h

```diff
@@ -9,6 +9,12 @@
 #include "ToneAlarmDriver.h"
 #include "Semaphores.h"
 
+enum hw_type {
+    HARDWARE_TYPE_BEBOP = 0,
+    HARDWARE_TYPE_BEBOP2,
+    NUM_HARDWARES,
+};
```

Comment:
I think you are mixing a generic implementation with a bebop-only implementation. If it's bebop-only, then you shouldn't export hw_type in the header, and you don't need _hw_names in the class. If it's generic, you should bring the get_rpi_version() method to the Util class and merge them.